webdataset/writer.py

```diff
@@ -341,9 +339,9 @@ def __exit__(self, exc_type, exc_val, exc_tb):
     def close(self):
         """Close the tar file."""
         self.tarstream.close()
-        if self.own_fileobj is not None:
```

Comment:
Here the current main branch code will never execute since self.own_fileobj is None
I believe it is reasonable to expect that an opened file will also be closed by the opening part, according to my patch